flaskoidc/__init__.py

```diff
@@ -76,65 +38,97 @@ def _before_request(self):
             token = request.args['access_token']
 
         if token:
-            validity = self.oidc.validate_token(token)
-            # This check True is required to make sure the validity is checked
-            if validity is True:
-                self.oidc.FLASKOIDC_ACCESS_TOKEN = token
-                return
-
-        # If not accepting a request, verify if the user is logged in
-        with self.app_context():
-            try:
-                if self.oidc.user_loggedin:
-                    access_token = self.oidc.get_access_token()
-                    assert access_token
-                    is_valid = self.oidc.validate_token(access_token)
-                    assert is_valid is True
-                return self.oidc.authenticate_or_redirect()
-            except (AssertionError, AttributeError):
-                # In case the session is forced logout from keycloak but still in
-                # the cookie, remove from cookie and try to login again
-                self.oidc.logout()
-                return self.oidc.authenticate_or_redirect()
+            token = json.loads(token)
+            if token.get("expires_at") <= _current_time:
+                LOGGER.exception("Token coming in request is expired")
+                abort(401)
```

Comment:
if the token is expired, why don't we try to get a new token via re-login?

Reply:
one more question, if a user is on some table page and then token is expired and we refresh the page , then will user be back on the same table page after login or on the home page?

Reply:
I failed to implement the `refresh_token` for now.  
Known issue here: https://github.com/verdan/flaskoidc#known-issues

Reply:
I believe you are referring in terms of the Amundsen project.
The table redirection will not work, as in the Amundsen project, we simply mute most of the exceptions/errors. Even if metadata returns 401, 
- frontend does not raise this 401. even if it does and return a redirect object to React
- the React app does not catch redirects or redirects at the moment. 

I have this in my pipeline, but will appreciate extra hands.

Reply:
Yes, I was facing this as well. I wanted to work on that too, but there are other priority things in my bucket as well.